### PR TITLE
Arsenal - Add Tools category and `moveOnOverwrite` parameter to `FUNC(addCustomRightPanelButton)`

### PR DESCRIPTION
**Pull request by LinkIsGrim**
**When merged this pull request will:**
- Add config property to sort items in CfgWeapons as tools.
- Add `moveOnOverwrite` parameter to `FUNC(addRightPanelButton)`. If true, button will be moved if its position is overridden (instead of deleted). Defaults to false.
- Add Tools Tab.
- Move Arsenal CBA Settings to initSettings.sqf.
- Add documentation for above and fix some typos.
- Deprecate Overheating item version of Spare Barrel.
- Improve some documentation.

Currently includes flashlights, Defusal Kit, clacklers, Notepad, Chemlight Shield, AtragMX, Kestrel, Fortify Tool, DAGR/MicroDAGR, Entrenching Tool, rangetables and range card, Wirecutters, SSWT kit, Spotting Scope, Mine Detector, Toolkit, Cellphone. Dead Man's Switch, HuntIR Monitor, and Map Tools.

### IMPORTANT

- [x] If the contribution affects [the documentation](https://github.com/acemod/ACE3/tree/master/docs), please include your changes in this pull request so the documentation will appear on the [website](https://ace3.acemod.org/).
- [x] [Development Guidelines](https://ace3.acemod.org/wiki/development/) are read, understood and applied.
- [x] Title of this PR uses our standard template `Component - Add|Fix|Improve|Change|Make|Remove {changes}`.

### CODE REVIEW

File: addons/arsenal/stringtable.xml [hunk 1]
```diff
@@ -1483,5 +1483,14 @@
             <Russian>Сохранить эмблему</Russian>
             <Spanish>Guardar Insignia</Spanish>
         </Key>
+        <Key ID="STR_ACE_Arsenal_toolsTab_displayName">
+            <English>Enable the tools tab</English>
+        </Key>
+        <Key ID="STR_ACE_Arsenal_toolsTab_description">
+            <English>Shows the tools tab in the left panel, containing items like toolkits, mine detectors, flashlights, etc.\nItems sorted to the Tools tab won't be shown in Misc. Items.</English>
```
Comment (Dystopian) [inline addons/arsenal/stringtable.xml:1490]:
```suggestion
            <English>Shows the tools tab in the right panel, containing items like toolkits, mine detectors, flashlights, etc.\nItems sorted to the Tools tab won't be shown in Misc. Items.</English>
```

**Review by Dystopian — COMMENTED** [review]
IMHO setting is not needed. Can't imagine situation when tools grouping can hurt.

File: addons/arsenal/stringtable.xml [hunk 2]
```diff
@@ -1483,5 +1483,14 @@
...
             <Spanish>Guardar Insignia</Spanish>
         </Key>
+        <Key ID="STR_ACE_Arsenal_toolsTab_displayName">
+            <English>Enable the tools tab</English>
+        </Key>
+        <Key ID="STR_ACE_Arsenal_toolsTab_description">
+            <English>Shows the tools tab in the right panel, containing items like toolkits, mine detectors, flashlights, etc.\nItems sorted to the Tools tab won't be shown in Misc. Items.</English>
+        </Key>
+        <Key ID="STR_ACE_Arsenal_toolsTab">
+            <English>Tools</English>
```
Comment (PabstMirror) [inline addons/arsenal/stringtable.xml:1493]:
```suggestion                          
            <English>Tools</English>
            <Czech>Nástroje</Czech>
            <German>Werkzeuge</German>
            <Russian>Инструменты</Russian>
            <Polish>Narzędzia</Polish>
            <Italian>Strumenti</Italian>
            <Spanish>Herramientas</Spanish>
            <French>Outils</French>
            <Chinese>工具</Chinese>
            <Japanese>ツール</Japanese>
            <Korean>도구</Korean>
            <Portuguese>Ferramentas</Portuguese>
            <Chinesesimp>工具</Chinesesimp>
            <Turkish>Araçlar</Turkish>
```

**Comment by LinkIsGrim** [comment]
> IMHO setting is not needed. Can't imagine situation when tools grouping can hurt.

People mad about their stuff not being in the usual place? I agree though.

**Comment by LinkIsGrim** [comment]
Is there anything else that should be sorted to Tools?

File: addons/arsenal/functions/fnc_addRightPanelButton.sqf [hunk 3]
```diff
@@ -6,21 +6,23 @@
...
  * Arguments:
- * 0: items only misc items <ARRAY of STRING>
- * 1: tooltip <STRING> (default: "")
- * 2: picture path <STRING> (default: QPATHTOF(data\iconCustom.paa))
- * 3: override a specific button (0-9) <NUMBER> (default: -1)
+ * 0: Items only misc items <ARRAY of STRING>
+ * 1: Tooltip <STRING> (default: "")
+ * 2: Picture path <STRING> (default: QPATHTOF(data\iconCustom.paa))
+ * 3: Override a specific button (0-9) <NUMBER> (default: -1)
+ * 4: Move button if its position is overriden <BOOL> (default: false)
```
Comment (rautamiekka) [inline addons/arsenal/functions/fnc_addRightPanelButton.sqf:13]:
```suggestion
 * 4: Move button if its position is overridden <BOOL> (default: false)
```

File: addons/arsenal/functions/fnc_addRightPanelButton.sqf [hunk 4]
```diff
@@ -6,21 +6,23 @@
...
  *
  * Example:
- * [["ACE_bloodIV_500", "ACE_Banana"], "MedicalStuff", "\z\ace\addons\arsenal\data\iconCustom.paa", 5] call ace_arsenal_fnc_addRightPanelButton
+ * [["ACE_bloodIV_500", "ACE_Banana"], "MedicalStuff", "\z\ace\addons\arsenal\data\iconCustom.paa", 5, false] call ace_arsenal_fnc_addRightPanelButton
  *
  * Public: Yes
 */
 
-params [["_items", [], [[]]], ["_tooltip", "", [""]], ["_picture", QPATHTOF(data\iconCustom.paa), [""]], ["_override", -1, [0]]];
+params [["_items", [], [[]]], ["_tooltip", "", [""]], ["_picture", QPATHTOF(data\iconCustom.paa), [""]], ["_override", -1, [0]], ["_keepIfOverriden", false, [false]]];
```
Comment (rautamiekka) [inline addons/arsenal/functions/fnc_addRightPanelButton.sqf:25]:
```suggestion
params [["_items", [], [[]]], ["_tooltip", "", [""]], ["_picture", QPATHTOF(data\iconCustom.paa), [""]], ["_override", -1, [0]], ["_keepIfOverridden", false, [false]]];
```

File: addons/arsenal/functions/fnc_addRightPanelButton.sqf [hunk 5]
```diff
@@ -49,7 +51,16 @@ if (_position >= 0 && _position <= 9) then {
...
+
+    // check if we're overwriting a button that's being force-kept
+    private _currentButtonInPosition = GVAR(customRightPanelButtons) select _position;
+    if (!isNil "_currentButtonInPosition") then {
+        _currentButtonInPosition params ["_cbItems", "_cbPicture", "_cbTooltip", "_cbKeep"];
+        if (_cbKeep) then {
+            [{_this call FUNC(addRightPanelButton)}, [_cbItems, _cbTooltip, _cbPicture, -1, _cbKeep]] call CBA_fnc_execNextFrame;
+        };
+    };
+    GVAR(customRightPanelButtons) set [_position, [_items apply {toLower _x}, _picture, _tooltip, _keepIfOverriden]];
```
Comment (rautamiekka) [inline addons/arsenal/functions/fnc_addRightPanelButton.sqf:63]:
```suggestion
    GVAR(customRightPanelButtons) set [_position, [_items apply {toLower _x}, _picture, _tooltip, _keepIfOverridden]];
```

File: docs/wiki/framework/arsenal-framework.md [hunk 6]
```diff
@@ -132,13 +132,27 @@ Examples:
...
-All of them are optional.
+Both of them are optional.
+
+### 3.2 Adding items to ACE's custom sub item categories
+
+ACE Arsenal includes a "Tools" custom sub item category by default:
+
+- `ACE_isTool`: Items with this property set to `1` will be sorted to the Tools Tab.
+
+ACE Medical Treatment and ACE Field Rations also have their own custom sub categories:
```
Comment (rautamiekka) [inline docs/wiki/framework/arsenal-framework.md]:
```suggestion
ACE Medical Treatment and ACE Field Rations also have their own custom subcategories:
```

File: docs/wiki/framework/arsenal-framework.md [hunk 7]
```diff
@@ -318,14 +332,16 @@ All are local.
 1  | Tooltip | String | Optional (default: `""`)
 2  | Picture path | String | Optional (default: `"\z\ace\addons\arsenal\data\iconCustom.paa"`)
 3  | Override a specific button | Number | Optional (default: `-1`)
+4  | Keep button if overriden | Bool | Optional (default: `false`)
```
Comment (rautamiekka) [inline docs/wiki/framework/arsenal-framework.md:335]:
```suggestion
4  | Keep button if overridden | Bool | Optional (default: `false`)
```

File: docs/wiki/framework/arsenal-framework.md [hunk 8]
```diff
@@ -318,14 +332,16 @@ All are local.
...
 
 Return Value:
 - successful: number of the slot (0-9)
 - error: -1
 
-This function creates a sub category under misc items in the ACE Arsenal.
-Only items that are listed under 'Misc Items' are available for sub categories.
+This function creates a sub category just above misc items in the ACE Arsenal.
+Only items that are listed under 'Misc. Items' or other sub categories are available for sub categories.
```
Comment (rautamiekka) [inline docs/wiki/framework/arsenal-framework.md:342]:
```suggestion
Only items that are listed under 'Misc. Items' or other sub-categories are available for subcategories.
```

File: docs/wiki/framework/arsenal-framework.md [hunk 9]
```diff
@@ -318,14 +332,16 @@ All are local.
...
 If the 'Override a specific button' argument is not used, the button will added at the bottom of the rest.
+If the 'Keep button if overriden' argument is used, the button will be moved to the bottom of the rest if its position is overriden.
```
Comment (rautamiekka) [inline docs/wiki/framework/arsenal-framework.md:344]:
```suggestion
If the 'Keep button if overriden' argument is used, the button will be moved to the bottom of the rest if its position is overridden.
```

**Review by rautamiekka — CHANGES_REQUESTED** [review]
Grammar/spelling fixes.

**Review by PabstMirror — APPROVED** [review]

**Review by johnb432 — COMMENTED** [review]
What happens/is supposed to happen if all custom panels are taken, all of them having `_moveOnOverwrite` enabled, and you add another panel with `_override != -1`?

Is the existing panel overwritten or is the panel rejected?

File: docs/wiki/framework/arsenal-framework.md [hunk 10]
```diff
@@ -402,14 +421,16 @@ All are local.
...
 Return Value:
 - Successful: Number of the slot (0-9)
 - Error: -1
 
-This function creates a sub category under misc. items in the ACE Arsenal.
-Only items that are listed under 'Misc. Items' are available for sub categories. 
+This function creates a sub-category just above misc items in the ACE Arsenal.
+Only items that are listed under 'Misc. Items' or other sub-categories are available for sub-categories.
 If the 'Override a specific button' argument is not used, the button will added at the bottom of the rest.
+If the 'Move button on overwrite' argument is used, the button will be moved to the bottom of the rest if its position is overriden.
```
Comment (johnb432) [inline docs/wiki/framework/arsenal-framework.md:433]: Looks like one of corrections you made earlier was reverted: `overriden` -> `overridden`

**Comment by LinkIsGrim** [comment]
> What happens/is supposed to happen if all custom panels are taken, all of them having `_moveOnOverwrite` enabled, and you add another panel with `_override != -1`?
> 
> Is the existing panel overwritten or is the panel rejected?

Existing panel is overwritten. Override wins.

**Pull request merged**